File: January2022/C++/can_place_flowers.cpp
created by @pranav108
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int temp=0,i=0,len=flowerbed.size();
        while(i<len&&flowerbed[i]==0)
            i++;
        if(i==len)
            return (len+1)/2>=n;
        n-=i/2;
        for(;i<len&&n>0;i++){
            if(flowerbed[i]){
                n-=temp>2?(temp-1)/2:0;
                    temp=0;
            }else
                temp++;
        }
        n-=temp/2;
        return n<=0;
    }
};

File: March2023/Java/CanPlaceFlowers.java
// @saorav21994

class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int l = flowerbed.length;
        for (int i = 0; i < l; i++) {
            if (n == 0)
                return true;
            boolean canPlace = true;
            if (flowerbed[i] == 0) {
                if (i > 0 && flowerbed[i-1] != 0)
                    canPlace = false;
                if (i < l-1 && flowerbed[i+1] != 0)
                    canPlace = false;
                if (canPlace) {
                    n -= 1;
                    flowerbed[i] = 1;
                }
            }

        }
        
        return (n == 0);
    }
}
